BiweeklyMeetings/2022-09-06.md

```diff
@@ -0,0 +1,353 @@
+
+**Matthias** -: But this is kind of the nature of my questions because you have two options also regarding your address. You can either modify the existing types or add new types that will include scripts, which means that you will still have the ability to use the old address type that does not include the script namespace. So I think this makes it slightly easier to implement in a non-breaking manner.
+
+**Sebastien** -: And I said this back in the CIP. So basically, there are two ways. There's one way of allocating new types, like the [inaudible 00:43:08] numbers, and the other way is to extend the existing types. We can extend the existing types without introducing a new number in the header, and this will work except for pointer addresses. The reason why is, because if you change the existing types, you can differentiate between them based on the length of the address. If it's one length, it's one standard. If it's another length, it's a different standard. This doesn't work for pointer addresses, because they have an undefined length, so there's not really a good way to parse them.
+
+**Matthias** -: Maybe there is a CIP to make here to suggest the removal of pointer addresses, since I think we are amongst the two individuals that have been using them on the chain, the only two. So yeah, that will simplify a bit the things. Okay. Should we get that CIP for review next week? Is the discussion still ongoing? Or I see mostly happened months ago, but no, no-
+
+**Sebastien** -: No, I don't think there's any ongoing discussion anymore. I think everybody has said their thoughts on the proposed path. I think some of the stuff has not been decided for the CIP 0039, the language annotated addresses. Michael Peyton Jones and Jared wanted to try and figure out how we can make this an on-chain construct that is actually validated by the blockchain itself. But then you need some way for the blockchain to know ahead of time whether not an address is a native script or a Plutus script. Or the other option is that if you don't validate this ahead of time, but then if somebody sends to the wrong script type, they can never spend from it, which is also not great.
+
+**Matthias** -: So it could be done fully optionally. So if you specify a type and you provide the script part of the transaction, then the ledger validates it. And if it doesn't validate, then it fails. So as a client, you have a way to sort of enforce that there is a ledger validation on top, but if you don't provide it, then yes, the ledger has no way to validate it. Then it's just blindly accepts it, like it does now.
+
+**Sebastien** -: Yeah. And the question is whether not this validation should have been when you send to the address or when you spend from the address.
+**Matthias** -: Well, I would say if you can do it when you send to the address, that's where the interest is at this stage, because that's where I think it can happen right now.
```

Comment:
```suggestion

**Matthias** -: Well, I would say if you can do it when you send to the address, that's where the interest is at this stage, because that's where I think it can happen right now.
```